framework/util/to_string.h

```diff
@@ -240,7 +240,7 @@ inline void JSONEscape(const char* cstr, std::string& escaped)
     if (cstr)
     {
         char c;
-        while (c = *cstr++)
+        while ((c = *cstr++))
```

Comment:
How about moving `char c` inside the while, like `while(char c = *str++)`?  That doesn't give me a warning for clang in Compiler Explorer and it moves `c` into a tighter scope.

Reply:
sounds good